### PR TITLE
test: Don't cache virt-builder templates during a GitHub task

### PR DESCRIPTION
**Pull request by mvollmer**
It might be too small.

### CODE REVIEW

File: test/guest/debian-unstable.bootstrap [hunk 1]
```diff
@@ -7,7 +7,14 @@ BASE=$(dirname $0)
...
 arch=$2
 
+# Avoid caching the templates in /tmp, as it might be too small.
+#
+if [ "$XDG_CACHE_HOME" == "/tmp" ]; then
+    virt_builder_caching="--no-cache"
+fi
+
 virt-builder debian-8 \
+             $virt_builder_caching
```
Comment (dperpeet) [inline test/guest/debian-unstable.bootstrap]: missing `\`

**Comment by dperpeet** [comment]
this feels very "ugly"

**Comment by mvollmer** [comment]
> this feels very "ugly"

The alternative it to remove `ENV XDG_CACHE_HOME /tmp` from https://github.com/cockpit-project/cockpituous/blob/master/verify/Dockerfile and deploying new images to all verify machines.

File: test/guest/debian-unstable.bootstrap (hunk 1 above)
Comment (mvollmer) [inline test/guest/debian-unstable.bootstrap]: Fixed.

**Comment by dperpeet** [comment]
> The alternative it to remove ENV XDG_CACHE_HOME /tmp from https://github.com/cockpit-project/cockpituous/blob/master/verify/Dockerfile and deploying new images to all verify machines.

It seems cleaner, but lacking comments I'm not sure what the effect of that would be. I seem to remember that tests were failing before. Why don't we just regularly restart the containers?

**Comment by mvollmer** [comment]
> Why don't we just regularly restart the containers?

This wouldn't help here.  /tmp is only 64M and not a sinle template would fit in it.

**Comment by dperpeet** [comment]
I feel that we should either check for available space (instead of path names) or, and I prefer this, the build environment should make sure that paths are set in a way that makes the builds possible.

**Comment by stefwalter** [comment]
I think avoiding the virt-builder cache during auto image creation is a good thing to do. Can we do it explicitly?

**Comment by mvollmer** [comment]
> I think avoiding the virt-builder cache during auto image creation is a good thing to do. Can we do it explicitly?

We can check whether we are running inside a container and avoid the cache in that case.  Or are you thinking of passing an explicit environment variable via the Dockerfile, like `DONT_USE_VIRT_BUILDER_CACHE=yes`?

**Comment by mvollmer** [comment]
bot: Image refresh for fedora-22

**Comment by cockpituous** [comment]
Image creation for fedora-22 in process on host37-rack09.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-fedora-22-81d1e773b9b4dc051547e6b63495563365a6afd4/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-fedora-22-81d1e773b9b4dc051547e6b63495563365a6afd4

**Comment by stefwalter** [comment]
> We can check whether we are running inside a container and avoid the cache in that case. Or are you thinking of passing an explicit environment variable via the Dockerfile, like DONT_USE_VIRT_BUILDER_CACHE=yes?

An environment variable would be appropriate, but can we set it while running github-task (or similar)? It seems that during any automated image creation we should avoid the local caches, right?

**Comment by mvollmer** [comment]
bot: Image refresh for fedora-22

**Comment by mvollmer** [comment]
> An environment variable would be appropriate, but can we set it while running github-task (or similar)?

Yes, that works, too.

**Comment by cockpituous** [comment]
Image creation for fedora-22 in process on host32-rack06.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-fedora-22-2913d88f53f7f03e96eea25e92eec2450bf8690e/

**Comment by cockpituous** [comment]
Image creation failed.

**Comment by mvollmer** [comment]
> Image creation failed.

This has failed because it was still using `github-task` from master.  I have tested this manually and virt-buildet gets the --no-cache option as expected.

**Comment by stefwalter** [comment]
> This has failed because it was still using github-task from master. I have tested this manually and virt-builder gets the --no-cache option as expected.

This is an issue that we need to come up with a better solution to. Not blocking this bug though.

**Pull request closed without merging**